Build failure with latest nightly

Compiler version:
`rustc 1.27.0-nightly (eeea94c11 2018-04-06)`

Errors:
```
error[E0609]: no field `identifier` on type `&syntax::ast::PathSegment`
   --> clippy_lints/src/utils/mod.rs:240:25
    |
240 |         .all(|(a, b)| a.identifier.name == *b)
    |                         ^^^^^^^^^^

error[E0599]: no method named `name` found for type `&syntax::ast::MetaItem` in the current scope
  --> clippy_lints/src/utils/conf.rs:16:16
   |
16 |         if arg.name() == "conf_file" {
   |                ^^^^

error[E0609]: no field `name` on type `&syntax::ast::MetaItem`
   --> clippy_lints/src/utils/author.rs:474:65
    |
474 |                 ast::NestedMetaItemKind::MetaItem(ref it) => it.name == "author",
    |                                                                 ^^^^

error[E0599]: no method named `name` found for type `&syntax::ast::MetaItem` in the current scope
   --> clippy_lints/src/attrs.rs:135:27
    |
135 |                     if mi.name() == "since";
    |                           ^^^^

error[E0599]: no method named `name` found for type `&syntax::ast::MetaItem` in the current scope
   --> clippy_lints/src/attrs.rs:331:28
    |
331 |         mi.is_word() && mi.name() == expected
    |                            ^^^^

error[E0609]: no field `span` on type `syntax::ast::Lifetime`
  --> clippy_lints/src/const_static_lifetime.rs:59:46
   |
59 |                                     lifetime.span,
   |                                              ^^^^ unknown field
   |
   = note: available fields are: `id`, `ident`

error[E0609]: no field `name` on type `syntax::ast::Variant_`
   --> clippy_lints/src/enum_variants.rs:122:14
    |
122 |     var.node.name.name.as_str()
    |              ^^^^ unknown field
    |
    = note: available fields are: `ident`, `attrs`, `data`, `disr_expr`

error[E0609]: no field `span` on type `&syntax::ast::TyParam`
   --> clippy_lints/src/misc_early.rs:198:28
    |
198 |                         ty.span,
    |                            ^^^^

error[E0609]: no field `identifier` on type `&syntax::ast::PathSegment`
   --> clippy_lints/src/misc_early.rs:212:18
    |
212 |                 .identifier
    |                  ^^^^^^^^^^

error[E0609]: no field `node` on type `syntax::ast::Ident`
   --> clippy_lints/src/misc_early.rs:271:41
    |
271 |                 let arg_name = sp_ident.node.to_string();
    |                                         ^^^^ unknown field
    |
    = note: available fields are: `name`, `span`

error[E0609]: no field `node` on type `syntax::ast::Ident`
   --> clippy_lints/src/misc_early.rs:337:33
    |
337 |                     if sp_ident.node == (&path.segments[0]).identifier {
    |                                 ^^^^ unknown field
    |
    = note: available fields are: `name`, `span`

error[E0609]: no field `identifier` on type `&syntax::ast::PathSegment`
   --> clippy_lints/src/misc_early.rs:337:61
    |
337 |                     if sp_ident.node == (&path.segments[0]).identifier {
    |                                                             ^^^^^^^^^^

error[E0609]: no field `node` on type `syntax::ast::Ident`
   --> clippy_lints/src/non_expressive_names.rs:107:69
    |
107 |             PatKind::Ident(_, id, _) => self.check_name(id.span, id.node.name),
    |                                                                     ^^^^ unknown field
    |
    = note: available fields are: `name`, `span`

error[E0308]: mismatched types
   --> clippy_lints/src/returns.rs:115:43
    |
115 |             if let ast::PatKind::Ident(_, Spanned { node: id, .. }, _) = local.pat.node;
    |                                           ^^^^^^^^^^^^^^^^^^^^^^^^ expected struct `syntax::ast::Ident`, found struct `syntax::codemap::Spanned`
    |
    = note: expected type `syntax::ast::Ident`
               found type `syntax::codemap::Spanned<_>`

error[E0609]: no field `identifier` on type `&syntax::ast::PathSegment`
  --> clippy_lints/src/unsafe_removed_from_name.rs:52:18
   |
52 |                 .identifier;
   |                  ^^^^^^^^^^

error: aborting due to 15 previous errors

Some errors occurred: E0308, E0599, E0609.
For more information about an error, try `rustc --explain E0308`.
error: Could not compile `clippy_lints`.
```

--- a/clippy_lints/src/attrs.rs
+++ b/clippy_lints/src/attrs.rs
@@ -132,7 +132,7 @@ impl<'a, 'tcx> LateLintPass<'a, 'tcx> for AttrPass {
                 if_chain! {
                     if let NestedMetaItemKind::MetaItem(ref mi) = item.node;
                     if let MetaItemKind::NameValue(ref lit) = mi.node;
-                    if mi.name() == "since";
+                    if mi.ident.name == "since";
                     then {
                         check_semver(cx, item.span, lit);
                     }
@@ -328,7 +328,7 @@ fn check_semver(cx: &LateContext, span: Span, lit: &Lit) {
 
 fn is_word(nmi: &NestedMetaItem, expected: &str) -> bool {
     if let NestedMetaItemKind::MetaItem(ref mi) = nmi.node {
-        mi.is_word() && mi.name() == expected
+        mi.is_word() && mi.ident.name == expected
     } else {
         false
     }

--- a/clippy_lints/src/const_static_lifetime.rs
+++ b/clippy_lints/src/const_static_lifetime.rs
@@ -56,7 +56,7 @@ impl StaticConst {
                                 span_lint_and_then(
                                     cx,
                                     CONST_STATIC_LIFETIME,
-                                    lifetime.span,
+                                    lifetime.ident.span,
                                     "Constants have by default a `'static` lifetime",
                                     |db| {
                                         db.span_suggestion(ty.span, "consider removing `'static`", sugg);

--- a/clippy_lints/src/enum_variants.rs
+++ b/clippy_lints/src/enum_variants.rs
@@ -119,7 +119,7 @@ impl LintPass for EnumVariantNames {
 }
 
 fn var2str(var: &Variant) -> InternedString {
-    var.node.name.name.as_str()
+    var.node.ident.name.as_str()
 }
 
 /// Returns the number of chars that match from the start

--- a/clippy_lints/src/misc_early.rs
+++ b/clippy_lints/src/misc_early.rs
@@ -195,7 +195,7 @@ impl EarlyLintPass for MiscEarly {
                     span_lint(
                         cx,
                         BUILTIN_TYPE_SHADOW,
-                        ty.span,
+                        ty.ident.span,
                         &format!("This generic shadows the built-in type `{}`", name),
                     );
                 }
@@ -209,7 +209,7 @@ impl EarlyLintPass for MiscEarly {
             let type_name = npat.segments
                 .last()
                 .expect("A path must have at least one segment")
-                .identifier
+                .ident
                 .name;
 
             for field in pfields {
@@ -267,8 +267,8 @@ impl EarlyLintPass for MiscEarly {
         let mut registered_names: HashMap<String, Span> = HashMap::new();
 
         for arg in &decl.inputs {
-            if let PatKind::Ident(_, sp_ident, None) = arg.pat.node {
-                let arg_name = sp_ident.node.to_string();
+            if let PatKind::Ident(_, ident, None) = arg.pat.node {
+                let arg_name = ident.name.to_string();
 
                 if arg_name.starts_with('_') {
                     if let Some(correspondence) = registered_names.get(&arg_name[1..]) {
@@ -328,13 +328,13 @@ impl EarlyLintPass for MiscEarly {
                 if let StmtKind::Local(ref local) = w[0].node;
                 if let Option::Some(ref t) = local.init;
                 if let ExprKind::Closure(_, _, _, _, _) = t.node;
-                if let PatKind::Ident(_, sp_ident, _) = local.pat.node;
+                if let PatKind::Ident(_, ident, _) = local.pat.node;
                 if let StmtKind::Semi(ref second) = w[1].node;
                 if let ExprKind::Assign(_, ref call) = second.node;
                 if let ExprKind::Call(ref closure, _) = call.node;
                 if let ExprKind::Path(_, ref path) = closure.node;
                 then {
-                    if sp_ident.node == (&path.segments[0]).identifier {
+                    if ident == (&path.segments[0]).ident {
                         span_lint(
                             cx,
                             REDUNDANT_CLOSURE_CALL,

--- a/clippy_lints/src/non_expressive_names.rs
+++ b/clippy_lints/src/non_expressive_names.rs
@@ -104,7 +104,7 @@ struct SimilarNamesNameVisitor<'a: 'b, 'tcx: 'a, 'b>(&'b mut SimilarNamesLocalVi
 impl<'a, 'tcx: 'a, 'b> Visitor<'tcx> for SimilarNamesNameVisitor<'a, 'tcx, 'b> {
     fn visit_pat(&mut self, pat: &'tcx Pat) {
         match pat.node {
-            PatKind::Ident(_, id, _) => self.check_name(id.span, id.node.name),
+            PatKind::Ident(_, ident, _) => self.check_name(ident.span, ident.name),
             PatKind::Struct(_, ref fields, _) => for field in fields {
                 if !field.node.is_shorthand {
                     self.visit_pat(&field.node.pat);

--- a/clippy_lints/src/returns.rs
+++ b/clippy_lints/src/returns.rs
@@ -1,6 +1,6 @@
 use rustc::lint::*;
 use syntax::ast;
-use syntax::codemap::{Span, Spanned};
+use syntax::codemap::Span;
 use syntax::visit::FnKind;
 
 use utils::{in_external_macro, in_macro, match_path_ast, snippet_opt, span_lint_and_then, span_note_and_lint};
@@ -112,9 +112,9 @@ impl ReturnPass {
             if local.ty.is_none();
             if !local.attrs.iter().any(attr_is_cfg);
             if let Some(ref initexpr) = local.init;
-            if let ast::PatKind::Ident(_, Spanned { node: id, .. }, _) = local.pat.node;
+            if let ast::PatKind::Ident(_, ident, _) = local.pat.node;
             if let ast::ExprKind::Path(_, ref path) = retexpr.node;
-            if match_path_ast(path, &[&id.name.as_str()]);
+            if match_path_ast(path, &[&ident.name.as_str()]);
             if !in_external_macro(cx, initexpr.span);
             then {
                     span_note_and_lint(cx,

--- a/clippy_lints/src/unsafe_removed_from_name.rs
+++ b/clippy_lints/src/unsafe_removed_from_name.rs
@@ -49,7 +49,7 @@ fn check_use_tree(use_tree: &UseTree, cx: &EarlyContext, span: &Span) {
                 .segments
                 .last()
                 .expect("use paths cannot be empty")
-                .identifier;
+                .ident;
             unsafe_to_safe_check(old_name, new_name, cx, span);
         }
         UseTreeKind::Simple(None) |

--- a/clippy_lints/src/utils/author.rs
+++ b/clippy_lints/src/utils/author.rs
@@ -471,7 +471,7 @@ fn has_attr(attrs: &[Attribute]) -> bool {
     attrs.iter().any(|attr| {
         attr.check_name("clippy") && attr.meta_item_list().map_or(false, |list| {
             list.len() == 1 && match list[0].node {
-                ast::NestedMetaItemKind::MetaItem(ref it) => it.name == "author",
+                ast::NestedMetaItemKind::MetaItem(ref it) => it.ident.name == "author",
                 ast::NestedMetaItemKind::Literal(_) => false,
             }
         })

--- a/clippy_lints/src/utils/conf.rs
+++ b/clippy_lints/src/utils/conf.rs
@@ -13,7 +13,7 @@ pub fn file_from_args(
     args: &[codemap::Spanned<ast::NestedMetaItemKind>],
 ) -> Result<Option<path::PathBuf>, (&'static str, codemap::Span)> {
     for arg in args.iter().filter_map(|a| a.meta_item()) {
-        if arg.name() == "conf_file" {
+        if arg.ident.name == "conf_file" {
             return match arg.node {
                 ast::MetaItemKind::Word | ast::MetaItemKind::List(_) => {
                     Err(("`conf_file` must be a named value", arg.span))

--- a/clippy_lints/src/utils/mod.rs
+++ b/clippy_lints/src/utils/mod.rs
@@ -237,7 +237,7 @@ pub fn match_path_ast(path: &ast::Path, segments: &[&str]) -> bool {
         .iter()
         .rev()
         .zip(segments.iter().rev())
-        .all(|(a, b)| a.identifier.name == *b)
+        .all(|(a, b)| a.ident.name == *b)
 }
 
 /// Get the definition associated to a path.

